File: eAutoShop/eAutoShop.Services/Database/AutoShopService.cs
using System;
using System.Collections.Generic;

namespace eAutoShop.Services.Database;

public partial class AutoShopService
{
    public int Id { get; set; }

    public string Name { get; set; } = null!;

    public double Price { get; set; }

    public double Discount { get; set; }

    public double DiscountedPrice { get; set; }

    public byte[] Image { get; set; } = null!;

    public string Description { get; set; } = null!;

    public int ServiceTypeId { get; set; }

    public string State { get; set; } = null!;

    public string? Details { get; set; }

    public TimeOnly Duration { get; set; }

    public virtual ICollection<AppointmentDetail> AppointmentDetails { get; set; } = new List<AppointmentDetail>();

    public virtual ServiceType ServiceType { get; set; } = null!;
}

File: eAutoShop/eAutoShop.UI/eautoshop_mobile/lib/models/autoshop_service/autoshop_service.dart
import 'package:json_annotation/json_annotation.dart';

part 'autoshop_service.g.dart';

@JsonSerializable()
class AutoShopService {
  int id;
  int serviceTypeId;
  String serviceTypeName;
  String name;
  double price;
  double discount;
  double discountedPrice;
  String state;
  String? imageData;
  String description;
  String? details;
  String duration;

  AutoShopService({
    required this.id,
    required this.serviceTypeId,
    required this.serviceTypeName,
    required this.name,
    required this.price,
    required this.discount,
    required this.discountedPrice,
    required this.state,
    this.imageData,
    required this.description,
    this.details,
    required this.duration,
  });

  factory AutoShopService.fromJson(Map<String, dynamic> json) =>
      _$AutoShopServiceFromJson(json);

  Map<String, dynamic> toJson() => _$AutoShopServiceToJson(this);
}
